### PR TITLE
Add 4chan thread image downloader

### PR DESCRIPTION
**Pull request by xanthisafk**
### Describe your change:

Adds an image downloader for 4chan thread.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/download_images_from_4chan.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    results = soup.find_all(class_="fileText")
+
+    links = []
+
+    for i in results:
+        k = i.find("a")
+        links.append("https://" + k["href"][2:])
+
+    return links
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:33]:
```suggestion
    return [f"https://{result.find('a')['href'][2:]}" for result in results]
```

File: web_programming/download_images_from_4chan.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        links: list containing all image URLs
+    
+    Returns:
+        None
+    """
+
+    print("Saving links...")
+
+    txt = ""
+    for i in links:
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:50]:
```suggestion
    for link in links:
```
Use `i` for integers.

File: web_programming/download_images_from_4chan.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        txt += i + "\n"
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:51]: Please use `str.join()` like `txt = "\n".join(links)`

File: web_programming/download_images_from_4chan.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    with open("links.txt", "w+") as f:
+        f.write(txt)
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:54]: Let's do this the modern way with https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text

File: web_programming/download_images_from_4chan.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    Entry point for program
+
+    Returns:
+        None
+    """
+    url:str = input("URL of 4chan thread: ")
+    _save:bool = (
+        True
+        if input("Save links? (overwrites [links.txt] if exists)[No]: ")
+        in ["y", "yes", "true"]
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:93]:
```suggestion
        in ("y", "yes", "true")
```
Tuples take up less memory than lists and they tell the reader that the contents will not change at runtime.

File: web_programming/download_images_from_4chan.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    """
+    Entry point for program
+
+    Returns:
+        None
+    """
+    url:str = input("URL of 4chan thread: ")
+    _save:bool = (
+        True
+        if input("Save links? (overwrites [links.txt] if exists)[No]: ")
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:92]:
```suggestion
        if input("Save links? (overwrites [links.txt] if exists)[No]: ").strip().lower()
```
What are the two reasons why is this safer?

File: web_programming/download_images_from_4chan.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+def main() -> None:
+    """
+    Entry point for program
+
+    Returns:
+        None
+    """
+    url:str = input("URL of 4chan thread: ")
+    _save:bool = (
+        True
```
Comment (cclauss) [inline web_programming/download_images_from_4chan.py:91]: `"y" in ("y", "yes", "true")` is already a `bool`.

**Comment by tianyizheng02** [comment]
Closing this PR because it's marked as a draft and has been inactive for almost a year. @xanthisafk, if you're still interested in this PR, please leave a comment to let me know and I can open it back up again.

**Pull request closed without merging**